src/components/spreadsheet/utils/cell-renderers.tsx

```diff
@@ -163,13 +163,33 @@ export const DefaultCellRenderer = (props: any) => {
 export const EllipsisCellRenderer = ({ value }: { value: any }) => {
     const textRef = useRef<any>(null);
     const [isEllipsisActive, setIsEllipsisActive] = useState(false);
-
-    useEffect(() => {
+    const checkEllipsis = () => {
         if (textRef.current) {
-            setIsEllipsisActive(textRef.current.scrollWidth > textRef.current.clientWidth);
+            const zoomLevel = window.devicePixelRatio;
+            const adjustedScrollWidth = textRef.current.scrollWidth / zoomLevel;
+            const adjustedClientWidth = textRef.current.clientWidth / zoomLevel;
+            setIsEllipsisActive(adjustedScrollWidth > adjustedClientWidth);
         }
+    };
+
+    useEffect(() => {
+        checkEllipsis();
     }, [value]);
 
+    useEffect(() => {
+        const handleResize = () => {
+            checkEllipsis();
+        };
+
+        window.addEventListener('resize', handleResize);
+        window.addEventListener('zoom', handleResize);
+
+        return () => {
+            window.removeEventListener('resize', handleResize);
+            window.removeEventListener('zoom', handleResize);
+        };
+    }, []);
```

Comment:
It is better to use ResizeObserver because it specifically observes changes in the size of cells, without relying on global window events

`const checkEllipsis = () => {
        if (textRef.current) {
            const zoomLevel = window.devicePixelRatio;
            const adjustedScrollWidth = textRef.current.scrollWidth / zoomLevel;
            const adjustedClientWidth = textRef.current.clientWidth / zoomLevel;
            setIsEllipsisActive(adjustedScrollWidth > adjustedClientWidth);
        }
    };

    useEffect(() => {
        checkEllipsis();
        const resizeObserver = new ResizeObserver(() => checkEllipsis());
        if (textRef.current) {
            resizeObserver.observe(textRef.current);
        }

        return () => {
            resizeObserver.disconnect();
        };
    }, [value]);`